hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDistributionSetTypeMapper.java

```diff
@@ -54,26 +55,30 @@ static DistributionSetType fromRequest(final EntityFactory entityFactory,
                 smsRest.getName(), smsRest.getDescription());
 
         // Add mandatory
-        smsRest.getMandatorymodules().stream().map(mand -> {
-            final SoftwareModuleType smType = softwareManagement.findSoftwareModuleTypeById(mand.getId());
+        if (!CollectionUtils.isEmpty(smsRest.getMandatorymodules())) {
```

Comment:
Better extract this block to a private method and remove the odd comment `Add mandatory`